blitz_api/models.py

```diff
@@ -232,6 +245,42 @@ def offer_free_membership(self, membership):
             )
         self.save()
 
+    def has_membership_active(self):
+        today = timezone.now().date()
+        return \
+            self.membership and \
+            self.membership_end and \
+            self.membership_end >= today
+
+    def has_to_receive_first_notify(self):
+        if self.has_membership_active():
+            today = timezone.now().date()
+            date_to_notify = self.membership_end - relativedelta(days=28)
+            # check if last notification between today and last week
+            if self.membership_end_notification:
+                already_notify = \
+                    today >= \
+                    self.membership_end_notification.date() >= \
+                    today - relativedelta(weeks=1)
+            else:
+                already_notify = False
+            # To prevent spam, we need to haven't send a
+            # notification in the last week and be at 28 days from the end
+            return date_to_notify == today and not already_notify
+
+    def check_and_notify_renew_membership(self):
+
+        if self.has_to_receive_first_notify():
+            services.notify_user_of_renew_membership(
+                user=self,
+                first_name=self.first_name,
+                membership_end=self.membership_end.strftime('%Y-%m-%d')
+            )
+            self.membership_end_notification = timezone.now()
+            self.save()
```

Comment:
In order to never forgot to update the `membership_end_notification` it could be better to add the update in the function `notify_user_of_renew_membership()` and add a new name param with True as default

ex: 
```
def notify_user_of_renew_membership(log=True):
    [...]
```